### DISCUSSION TITLE
Bamboo - fastlane task failed

### DISCUSSION THREAD

**Discussion by Parigi65** [fastlane/fastlane#17455] [Q&A]
Hello @joshdholtz, everyone;
I'm writing to ask for help. I'm using fastlane task into bamboo to automate the deployment of an iOS app.

I've created the .p8 file for "app_store_connect_api_key"
I've filled the environment variables field into the fastlane task:
"LANG=UTF-8 LC_ALL=UTF-8 APP_STORE_CONNECT_API_KEY_KEY_ID=“xxxx” APP_STORE_CONNECT_API_KEY_ISSUER_ID=“xxxx” APP_STORE_CONNECT_API_KEY_KEY_FILEPATH="${bamboo.build.working.directory}/xx/AuthKey_xxx.p8" APP_STORE_CONNECT_API_KEY_DURATION=1200 APP_STORE_CONNECT_API_KEY_IN_HOUSE=false"

Every other fastlane operation like "build" is ok.
When the plan arrives to:
Starting task 'Build and upload to Testflight' of type 'com.atlassian.bamboo.plugins.xcode.bamboo-xcode-plugin:fastlaneTaskType'

The error:
Failing task since return code of [/var/folders/7p/fbd7gb613c55gkt9386dlx5w0000gp/T/epb2966327233818768158.sh fastlane ios beta] was -1 while expected 0

Why?
Please help me.

**Comment by Parigi65** [comment]
An image of the fastlane task inside bamboo.
<img width="763" alt="Schermata 2020-10-16 alle 18 29 25" src="https://user-images.githubusercontent.com/11082140/96284278-a5df5c80-0fdd-11eb-9905-39e57c8e3293.png">

**Comment by rogerluan** [comment]
Hi @Parigi65, it's very hard to take any conclusion based on the information you gave. Looks like Bamboo abstracts away certain steps from you, so that you're forced to use their way of doing things. We will need a stacktrace of the crash to be able to identify the error. Your best shot would be to ask help directly from Bamboo customer support probably.

Don't forget to update _fastlane_ to the latest version before checking again. Maybe your issue was resolved already.
